java-frontend/src/main/java/org/sonar/java/se/checks/NullDereferenceCheck.java

```diff
@@ -150,9 +151,12 @@ private void reportIssue(SymbolicValue currentVal, Tree syntaxNode, ExplodedGrap
   @Nullable
   private static Symbol dereferencedSymbol(Tree syntaxNode) {
```

Comment:
This can be now implemented by using symbol on the stack, using the `node` and `currentVal` in `reportIssue`
```
ProgramState.SymbolicValueSymbol symbolicValueSymbol = node.programState.peekValueSymbol();
    Symbol dereferencedSymbol = symbolicValueSymbol.symbolicValue() == currentVal ? symbolicValueSymbol.symbol() : null;
```